packages/vue-components/src/components/OmegaForm/OmegaWrapper.vue

```diff
@@ -90,16 +90,53 @@ const props = defineProps<
       schema: S.Schema<To, From, never>
     }
   )
->()
+  & (
+    | {
+      isLoading?: undefined
+      onSubmit?: FormProps<From, To>["onSubmit"]
+    }
+    | {
+      isLoading: boolean
+      onSubmit: (data: To) => void
+    }
+  )
 
-const localForm = computed(() => {
-  if (props.form || !props.schema) {
-    return undefined
-  }
-  return useOmegaForm<From, To>(props.schema, props, props.omegaConfig)
+const props = withDefaults(defineProps<OmegaWrapperProps>(), {
+  isLoading: undefined
 })
 
-const formToUse = computed(() => props.form ?? localForm.value!)
+const instance = getCurrentInstance()
+
+// we prefer to use the standard abstraction in Vue which separates props (going down) and event emits (going back up)
+// so if isLoading + @submit are provided, we wrap them into a Promise, so that TanStack Form can properly track the submitting state.
+// we use this approach because it means we can keep relying on the built-in beaviour of TanStack Form, and we dont have to re-implement/keep in sync/break any internals.
+const eventOnSubmit: FormProps<From, To>["onSubmit"] = ({ value }) => {
+  new Promise<void>((resolve) => {
+    instance!.emit("submit", value)
+    // even if the emit would be immediately handled, prop changes are not published/received immediately.
+    // so we have to wait for the prop to change to true, and back to false again.
+    const handle = watch(() => props.isLoading, (v) => {
+      if (v) return
+      resolve()
+      handle.stop()
+    })
+  })
+}
+
+const localForm = props.form || !props.schema
```

Comment:
this is now no longer reactive, so when props change, there's no reaction.
not that it happens a lot I guess :)